backend/src/email/routes/tests/email-transactional.routes.test.ts

```diff
@@ -722,7 +724,306 @@ describe('POST /transactional/email/send', () => {
+        ...acceptedMessage,
+        userId: user.id,
+      } as unknown as EmailMessageTransactional)
+    }
+    for (let i = 0; i < 5; i++) {
+      await EmailMessageTransactional.create({
+        ...sentMessage,
+        userId: user.id,
+      } as unknown as EmailMessageTransactional)
+    }
+    const res = await request(app)
+      .get(`${endpoint}?status=delivered`) // case-insensitive
+      .set('Authorization', `Bearer ${apiKey}`)
+    expect(res.status).toBe(200)
+    expect(res.body.has_more).toBe(false)
+    expect(res.body.data.length).toBe(5)
+    res.body.data.forEach((message: EmailMessageTransactional) => {
+      expect(message.status).toBe(TransactionalEmailMessageStatus.Delivered)
+    })
+    const res2 = await request(app)
+      .get(`${endpoint}?status=aCcEPteD`)
+      .set('Authorization', `Bearer ${apiKey}`)
+    expect(res2.status).toBe(200)
+    expect(res2.body.has_more).toBe(false)
+    expect(res2.body.data.length).toBe(5)
+    res2.body.data.forEach((message: EmailMessageTransactional) => {
+      expect(message.status).toBe(TransactionalEmailMessageStatus.Accepted)
+    })
+    const res3 = await request(app)
+      .get(`${endpoint}?status=SENT`)
+      .set('Authorization', `Bearer ${apiKey}`)
+    expect(res3.status).toBe(200)
+    expect(res3.body.has_more).toBe(false)
+    expect(res3.body.data.length).toBe(5)
+    res3.body.data.forEach((message: EmailMessageTransactional) => {
+      expect(message.status).toBe(TransactionalEmailMessageStatus.Sent)
+    })
+    // duplicate status params should throw an error
+    const res4 = await request(app)
+      .get(`${endpoint}?status=SENT&status=ACCEPTED`)
+      .set('Authorization', `Bearer ${apiKey}`)
+    expect(res4.status).toBe(400)
+  })
+  test('created_at filter range should work', async () => {
+    const messages = []
+    const now = new Date()
+    for (let i = 0; i < 10; i++) {
+      const message = await EmailMessageTransactional.create({
+        ...deliveredMessage,
+        userId: user.id,
+        createdAt: new Date(now.getTime() - 100000 + i * 1000), // inserting in chronological order
+      } as unknown as EmailMessageTransactional)
+      messages.push(message)
+    }
+    const res = await request(app)
+      .get(
+        `${endpoint}?created_at[gte]=${messages[0].createdAt.toISOString()}&created_at[lte]=${messages[4].createdAt.toISOString()}`
+      )
+      .set('Authorization', `Bearer ${apiKey}`)
+    expect(res.status).toBe(200)
+    expect(res.body.has_more).toBe(false)
+    expect(res.body.data.length).toBe(5)
+
+    const res2 = await request(app)
+      .get(
+        `${endpoint}?created_at[gt]=${messages[0].createdAt.toISOString()}&created_at[lt]=${messages[4].createdAt.toISOString()}`
+      )
+      .set('Authorization', `Bearer ${apiKey}`)
+    expect(res2.status).toBe(200)
+    expect(res2.body.has_more).toBe(false)
+    expect(res2.body.data.length).toBe(4) // not sure why this is 4 and not 3
```

Comment:
honestly don't know why this is 4 and not 3 (sorry @stanleynguyen I know you said I should remove comments from the code, but I needed to include this to mark out the weird statement from the sea of tests... will remove once this is resolved)

Reply:
oops think it's a bug let me fix